### ISSUE TITLE
 Deprecated method: this method is going to be deleted. Use Fluent::MessagePackFactory.msgpack_unpacker

### ISSUE THREAD

**Issue by liorco3000** [awslabs/aws-fluent-plugin-kinesis#194]
Problem
We see hundreds of deprecation warnings per minute:
```
fuentd-agent-v2-jh5tn fluentd-agent 2020-01-30 12:18:59 +0000 [warn]: #0 Deprecated method: this method is going to be deleted. Use Fluent::MessagePackFactory.msgpack_unpacker
```
So i guess it related to this plugin we added recently.
Any fix for this ?

**Comment by mindw** [comment]
reference fix:
fluent-plugins-nursery/fluent-plugin-cloudwatch-logs#172

**Comment by eredi93** [comment]
I'm also getting this warning. @simukappu is it as simple as preparing a PR like the one @mindw shared in the comment above?

**Comment by simukappu** [comment]
Hi,
Thank you for your feedback. Which version of Fluentd are you using?
The class method *Fluent::MessagePackFactory.msgpack_unpacker* was added after Fluentd v1.8 [here](https://github.com/fluent/fluentd/commit/5324ee44676a1021dc94d1e23738653e5c0a8220). Deprecated log was also added [here](https://github.com/fluent/fluentd/commit/4209dcf8ed569578f71ca51658098e857542a72b) with this. However, [the latest td-agent v3.5.1 includes Fluentd v1.7.4](https://support.treasuredata.com/hc/en-us/articles/360001479187-The-td-agent-ChangeLog).
We would appreciate it if you could give us more feedback. Thanks!

**Comment by eredi93** [comment]
@simukappu I'm using the lastest version of fluentd 1.9.1
is the issue here the fact that it would break td-agent?
I'm just wondering if we could work on removing the deprecated message, but if it would break other standard deployments like td-agent then i'm happy to leave it as is until they upgrade the package to 1.8 or newer

**Comment by simukappu** [comment]
Hi @eredi93,
Could you test this patch whether you can remove the deprecated messages?
```bash
$ git clone https://github.com/awslabs/aws-fluent-plugin-kinesis.git
$ cd aws-fluent-plugin-kinesis
$ git checkout issue-194
$ gem build fluent-plugin-kinesis.gemspec

$ td-agent-gem install fluent-plugin-kinesis-3.2.0.gem
# Or
$ fluent-gem install fluent-plugin-kinesis-3.2.0.gem
```

**Comment by eredi93** [comment]
@simukappu i can confirm the deprecated messages are gone when using the patch 🙇

**Comment by simukappu** [comment]
Released as v3.2.1. Thank you!
https://rubygems.org/gems/fluent-plugin-kinesis/versions/3.2.1
